QMLComponents/controls/radiobuttonsgroupbase.cpp

```diff
@@ -33,50 +32,46 @@ RadioButtonsGroupBase::RadioButtonsGroupBase(QQuickItem* item)
 
 void RadioButtonsGroupBase::setUp()
 {
-	QList<RadioButtonBase* > buttons;
-	_getRadioButtons(this, buttons);
-	QVariant buttonGroup = property("buttonGroup");
-
-	connect(this,	&RadioButtonsGroupBase::clicked, this,	&RadioButtonsGroupBase::clickedSlot);
+	JASPControl::setUp();
+}
 
-	for (RadioButtonBase* button: buttons)
+void RadioButtonsGroupBase::registerRadioButton(RadioButtonBase* button)
+{
+	const QString& controlName = button->name();
+	if (controlName.isEmpty())
+		addControlError(tr("A RadioButton inside RadioButtonGroup element (name: %1) does not have any name").arg(name()));
+	else
 	{
-		const QString& controlName = button->name();
-		if (controlName.isEmpty())
-			addControlError(tr("A RadioButton inside RadioButtonGroup element (name: %1) does not have any name").arg(name()));
-		else
-		{
-			_buttons[controlName] = button;
-			if (button->property("checked").toBool())	setValue(controlName);
-			button->setProperty("buttonGroup", buttonGroup);
-		}
+		if (_buttons.size() == 0)
+			_setCheckedButton(button);
+		if (button->property("checked").toBool())
+			_setCheckedButton(button);
```

Comment:
To avoid checking radio buttons multiple times during the initialization of the RadioButtonGroup: if it is not yet initialized, don't set the checked button, and connect the initialized property to set the right checked button.